cloudimg/ms_azure.py

```diff
@@ -285,11 +287,15 @@ def filter_object_by_tags(self, tags):
                 blobs_list = container_client.find_blobs_by_tags(
                     filter_expression=filter_expression
                 )
-                try:
-                    return blobs_list.next()
-                except StopIteration:
-                    continue
-        return
+                for tag in tags.keys():
+                    log.debug("Filtering the blob with the tag: %s" % tag)
+                    try:
+                        blobs_list = blobs_list.next()
```

Comment:
Replacing `blobs_list` with the returned value from `.next()` is suspicious.  What are the types of the variables here? 

I think find_blobs_by_tags is this method, isn't it? https://learn.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.blobserviceclient?view=azure-python#azure-storage-blob-blobserviceclient-find-blobs-by-tags

If so, then isn't it like this:

- Initially, `blobs_list` is an instance of `ItemPaged[FilteredBlob]`.
- After the first iteration of `blobs_list = blobs_list.next()`, `blobs_list` is a `FilteredBlob` (because that's what `next` returns).
- At the second iteration, it crashes because `FilteredBlob` doesn't have a `next` method.

It seems buggy.  If it somehow works, could you explain a bit how? e.g. am I looking at the wrong docs?

Reply:
You're correct, this implementation is wrong...

It asks the `FilteredBlob` to perform a `next` when it doesn't exist.